### ISSUE TITLE
onDragStart event for markers

### ISSUE THREAD

**Issue by ahmed1490** [lelandrichardson/react-native-maps#71]
It would be nice to have the ability to change marker image or the css to zoom in the image on drag start and subsequently change it back on drag end.

A is dragging property to the custom marker is another case, to consider.

**Comment by ahmed1490** [comment]
```
        <MapView.Marker draggable
          coordinate={this.state.startMarker}
          onDragEnd={(e) => this.setState({ startMarker: e.nativeEvent.coordinate })}
          title="This is a title"
          description="This is a description"
          onSelect={console.log('onSelect', arguments)}
          onDrag={console.log('onDrag', arguments)}
          onDragStart={console.log('onDragStart', arguments)}
          onPress={console.log('onPress', arguments)}
        >
          <StartPin amount={99} startMarker={this.state.startMarker}/>
        </MapView.Marker>
```

all the console.log happens *after* i have dropped the marker.

**Comment by lelandrichardson** [comment]
@ahmed1490 Hmm. It's certainly supposed to work differently. Does the `onDragStart` logging ever actually get called?

Can you try listening to these events on the `<MapView />` component and see if the logging there is any different?

**Comment by ahmed1490** [comment]
I now tried..
```
        <MapView.Marker draggable
          coordinate={this.state.startMarker}
          // onSelect={console.log('onSelect', arguments)}
          // onDrag={console.log('onDrag', arguments)}
          // onDragStart={console.log('onDragStart', arguments)}
          // onPress={console.log('onPress', arguments)}
        >
          <StartPin amount={99} />
        </MapView.Marker>

        <MapView.Marker draggable
          coordinate={this.state.startMarker}
          onDrag={(e) => this.setState({ startMarker: e.nativeEvent.coordinate })}
        >
        </MapView.Marker>
```

both the components seem to be in sync while dragging.. 
But with console log.. i see all the changes.. only after i dropped back the marker

**Comment by christopherdro** [comment]
@ahmed1490 Your console logs are automatically getting invoked at render since your using `()` by passing in arguments. You'll have to call your console logs with a anonymous function.

```jsx
<MapView.Marker
  coordinate={this.state.a}
  onSelect={() => console.log('onSelect', arguments)}
  onDrag={() => console.log('onDrag', arguments)}
  onDragStart={() => console.log('onDragStart', arguments)}
  draggable
>
```

**Comment by ahmed1490** [comment]
Right..
Thank you :)

**Comment by christopherdro** [comment]
:+1:
